shell/platform/glfw/flutter_glfw.cc

```diff
@@ -315,6 +325,10 @@ static void SendPointerEventWithData(GLFWwindow* window,
     controller->pointer_currently_added = true;
   } else if (event_data.phase == FlutterPointerPhase::kRemove) {
     controller->pointer_currently_added = false;
+  } else if (event_data.phase == FlutterPointerPhase::kDown) {
```

Comment:
how do we keep track of which of the two button is down if we only use 1 boolean?

For example, if left-click down and right click down and you lifted left click, the controller->pointer_currently_down will become false

Reply:
I messed up the conditional here a bit in a clean up commit, it is supposed to be (buttons == 0). I think your question still applies though:

Basically this is really ugly, but the engine doesn't differentiate between left and right mouse buttons when setting the kDown and kUp states. The buttons field is basically just additional field for telling which buttons are currently down. This is basically a copy of what is happening in MacOS here:

https://github.com/flutter/engine/blob/master/shell/platform/darwin/macos/framework/Source/FlutterViewController.mm#L424

The kUp and kDown events here are only called if the stored buttons state moves from no buttons presses (buttons == 0) to any buttons pressed, or vice versa. 

So, in the case you describe - when the left click is lifted the SetEventPhaseFromCursorButtonState will trigger a kMove phase with the buttons being updated to just the right mouse button. This is obviously not entirely correct, but it seems like the best option and is a mirror of MacOS.

Reply:
Oh yes, I forgot about this, lifting one finger while pressing down another should issue a move, and as long as that is the case in linux embedding that should be fine. I think what you have right now is correct.